src/wp-includes/class-wp-theme-json.php

```diff
@@ -1006,6 +1006,27 @@ public function get_stylesheet( $types = array( 'variables', 'styles', 'presets'
 		return $stylesheet;
 	}
 
+	/**
+	 * Processes the CSS, to apply nesting.
+	 *
+	 * @param string $css      The CSS to process.
+	 * @param string $selector The selector to nest.
+	 *
+	 * @return string The processed CSS.
+	 */
+	public function process_blocks_custom_css( $css, $selector ) {
```

Comment:
@aristath, I would revisit the visibility of this new method. We don't know if the method in its current state will be helpful outside of the block custom CSS processing.

Maybe we should ship this a private method since there's no risk of breaking BC.

cc @hellofromtonya

Reply:
Thank you @Mamaduka 👍 
Changed the visibility to `protected`, fixed the tests, and ported the change to Gutenberg in https://github.com/WordPress/gutenberg/pull/47725